### ISSUE TITLE
Rule proposal: Use `Number.isInteger`

### ISSUE THREAD

**Issue by jonahsnider** [sindresorhus/eslint-plugin-unicorn#1460]
Use [`Number.isInteger`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger).
Not really sure if there's any benefit to using it. Maybe it's more performant. Makes the code marginally cleaner.

## Fail

```js
number % 2 === 0;
```

## Pass

```js
Number.isInteger(number);
```

**Comment by sindresorhus** [comment]
> Makes the code marginally cleaner.

Marginally cleaner is an understatement. `number % 1 === 0;` is extremely unreadable and also fails if the passed in variable is not a number:

```js
let number = [['1']];

// ...

number % 1 === 0;
//=> true
```

Some more cases it could catch: https://stackoverflow.com/a/34848042/64949

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see the initial attempt and feedback in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1509
